src/anomalib/data/kolektor.py

```diff
@@ -0,0 +1,315 @@
+"""Kolektor Surface-Defect Dataset (CC BY-NC-SA 4.0).
+
+Description:
+    This script contains PyTorch Dataset, Dataloader and PyTorch
+        Lightning DataModule for the Kolektor Surface-Defect dataset.
+    The dataset can be found at https://www.vicos.si/resources/kolektorsdd/
+
+License:
+    Kolektor Surface-Defect dataset is released under the Creative Commons
+    Attribution-NonCommercial-ShareAlike 4.0 International License
+    (CC BY-NC-SA 4.0)(https://creativecommons.org/licenses/by-nc-sa/4.0/).
+
+Reference:
+    - Tabernik, Domen, Samo Šela, Jure Skvarč, and Danijel Skočaj.
+    "Segmentation-based deep-learning approach for surface-defect detection."
+    Journal of Intelligent Manufacturing 31, no. 3 (2020): 759-776.
+"""
+
+
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from urllib.request import urlretrieve
+from zipfile import ZipFile
+
+import albumentations as A
+import numpy as np
+from imageio.v2 import imread
+from pandas import DataFrame
+from sklearn.model_selection import train_test_split
+
+from anomalib.data.base import AnomalibDataModule, AnomalibDataset
+from anomalib.data.task_type import TaskType
+from anomalib.data.utils import (
+    InputNormalizationMethod,
+    Split,
+    TestSplitMode,
+    ValSplitMode,
+    get_transforms,
+)
+from anomalib.data.utils.download import hash_check
+
+logger = logging.getLogger(__name__)
+
+
+# Check if a mask shows defects
+def is_good(path):
```

Comment:
Not sure if this name is descriptive enough. Maybe your comment could be used as the name of the function such as `is_mask_anomalous or something similar?